### PR TITLE
API: dot correctness

### PR DESCRIPTION
**Pull request by zeha**
From and to API consumers we'll now always require/send names (and
content) with dots. To the backend, we'll always require/send without
dots. Some API tests now check the data written to the sqlite DB, too.

Incoming names are now checked against a restricted list of chars,
fixing #1437. The double dot case is taken care of by DNSName (and
we'll no longer report an ISE if DNSName parsing fails - we make
sure to parse all names in a try/except).

Cleanup leftovers from pre-DNSName times.

Turn auth api tests back on in travis.

### CODE REVIEW

**Comment by zeha** [comment]
Can't figure out why the "bind" zone import messes up the dots:

```
======================================================================
FAIL: test_import_zone_axfr (test_Zones.AuthZones)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/ch/Source/PowerDNS/pdns/regression-tests.api/test_Zones.py", line 343, in test_import_zone_axfr
    eq_zone_dict(data['records'], expected)
  File "/Users/ch/Source/PowerDNS/pdns/regression-tests.api/test_helper.py", line 63, in eq_zone_dict
    assert types_records == expected, "%r != %r" % (types_records, expected)
AssertionError: ...
-------------------- >> begin captured stdout << ---------------------
{'A': [{'content': u'82.94.213.34', 'name': u'powerdns.com.'}],
 'AAAA': [{'content': u'2001:888:2000:1d::2', 'name': u'powerdns.com.'}],
 'MX': [{'content': u'0 xs.powerdns.com'}],
 'NS': [{'content': u'powerdnssec2.ds9a.nl'},
        {'content': u'powerdnssec1.ds9a.nl'}],
 'SOA': [{'content': u'powerdnssec1.ds9a.nl ahu.ds9a.nl 1343746984 10800 3600 604800 10800'}]}
{'A': [{'content': '82.94.213.34', 'name': 'powerdns.com.'}],
 'AAAA': [{'content': '2001:888:2000:1d::2', 'name': 'powerdns.com.'}],
 'MX': [{'content': '0 xs.powerdns.com.'}],
 'NS': [{'content': 'powerdnssec1.ds9a.nl.'},
        {'content': 'powerdnssec2.ds9a.nl.'}],
 'SOA': [{'content': 'powerdnssec1.ds9a.nl. ahu.ds9a.nl. 1343746984 10800 3600 604800 10800'}]}

--------------------- >> end captured stdout << ----------------------

======================================================================
FAIL: test_import_zone_bind (test_Zones.AuthZones)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/ch/Source/PowerDNS/pdns/regression-tests.api/test_Zones.py", line 398, in test_import_zone_bind
    eq_zone_dict(data['records'], expected)
  File "/Users/ch/Source/PowerDNS/pdns/regression-tests.api/test_helper.py", line 63, in eq_zone_dict
    assert types_records == expected, "%r != %r" % (types_records, expected)
AssertionError: ...
-------------------- >> begin captured stdout << ---------------------
{'A': [{'content': u'192.168.0.3', 'name': u'bill.example.org.'},
       {'content': u'192.168.0.4', 'name': u'fred.example.org.'},
       {'content': u'192.168.0.1', 'name': u'ns1.example.org.'},
       {'content': u'192.168.0.2', 'name': u'www.example.org.'}],
 'CNAME': [{'content': u'www.example.org', 'name': u'ftp.example.org.'}],
 'MX': [{'content': u'10 mail.another.com'}],
 'NS': [{'content': u'ns1.example.org'}, {'content': u'ns2.smokeyjoe.com'}],
 'SOA': [{'content': u'ns1.example.org hostmaster.example.org 2002022401 10800 15 604800 10800'}]}
{'A': [{'content': '192.168.0.1', 'name': 'ns1.example.org.'},
       {'content': '192.168.0.2', 'name': 'www.example.org.'},
       {'content': '192.168.0.3', 'name': 'bill.example.org.'},
       {'content': '192.168.0.4', 'name': 'fred.example.org.'}],
 'CNAME': [{'content': 'www.example.org', 'name': 'ftp.example.org.'}],
 'MX': [{'content': '10 mail.another.com.'}],
 'NS': [{'content': 'ns1.example.org.'}, {'content': 'ns2.smokeyjoe.com.'}],
 'SOA': [{'content': 'ns1.example.org. hostmaster.example.org. 2002022401 10800 15 604800 10800'}]}

--------------------- >> end captured stdout << ----------------------
```

**Comment by Habbie** [comment]
The same issue exists generally:
```
pdnssec --config-dir=. --config-name=gsqlite3 check-zone test.com
[Warning] Parsed and original record content are not equal: *.test.test.com. IN CNAME 'server1.test.com' (Content parsed as 'server1.test.com.')
```

This is very high on my list of things to dig into.

**Comment by zeha** [comment]
With my last commit, all API tests pass, but the following changes:

```
../pdns/pdnssec --config-dir=. --config-name=bind check-zone test.com
[Warning] Parsed and original record content are not equal: test.com. IN SOA 'ns1.test.com ahu.example.com 2005092501 28800 7200 604800 86400' (Content parsed as 'ns1.test.com. ahu.example.com. 2005092501 28800 7200 604800 86400')
[Warning] Parsed and original record content are not equal: test.com. IN NS 'ns1.test.com' (Content parsed as 'ns1.test.com.')
...
```

to

```
../pdns/pdnssec --config-dir=. --config-name=bind check-zone test.com
[Warning] The record test.com. with type NS has a trailing dot in the content (ns1.test.com.). Your backend might not work well with this.
[Warning] The record test.com. with type NS has a trailing dot in the content (ns2.test.com.). Your backend might not work well with this.
```

Which might be bad at this point?

**Comment by zeha** [comment]
OTOH, `start-test-stop 5399 bind` still gives me `86 out of 86 (100.00%) tests passed, 71 were skipped`.

**Comment by Habbie** [comment]
Yes, sticking trailing dots in the database is bad. I have a WIP branch for fixing all the dot stuff, taking a slightly different approach. No time for it today though :(

**Comment by Habbie** [comment]
WIP branch no longer WIP, PR open at #2964

**Comment by zeha** [comment]
Reworked this - backends are now properly fed names/content without dot, while API consumers always get the dot.

File: docs/markdown/httpapi/api_spec.md [hunk 1]
```diff
@@ -567,7 +576,8 @@ zone\_metadata\_resource
       ]
     }
 
-Valid values for `<metadata_kind>` are specified in <http://doc.powerdns.com/domainmetadata.html>.
+Valid values for `<metadata_kind>` are specified in
+[`the domainmetadata documentation`](../authoritative/domainmetadata.md).
```
Comment (pieterlexis) [inline docs/markdown/httpapi/api_spec.md]: [the `domainmetadata` documentation]

**Comment by Habbie** [comment]
LGTM!

**Comment by zeha** [comment]
Squashed the fixes (cvstrac reference gone now, PacketCache FIXME has a note what to fix for)

**Pull request merged**